providers/src/airflow/providers/edge/cli/edge_command.py

```diff
@@ -94,6 +94,33 @@ def _hostname() -> str:
 def _pid_file_path(pid_file: str | None) -> str:
     return cli_utils.setup_locations(process=EDGE_WORKER_PROCESS_NAME, pid=pid_file)[0]
 
+def _write_pid_to_pidfile(pid_file_path):
+    """
+    Write PIDs for Edge Workers to disk,
+    ensuring that orphanded PID files from crashed instance are handled.
+    """
+    if pid_file_path.exists():
+        # Handle existing PID files on disk
+        logger.info("An existing PID file has been found: %s.", pid_file_path)
+        pid_stored_in_pid_file = read_pid_from_pidfile(pid_file_path)
+        if os.getpid() == pid_stored_in_pid_file:
+            raise SystemExit("A PID file has already been written")
+        else:
+            # PID file was written by dead or already running instance
+            logger.info("PID file does not belong to this process.")
+            if psutil.pid_exists(pid_stored_in_pid_file):
+                # case 1: another instance uses the same path for its PID file
+                raise SystemExit(
+                    f"The PID file {pid_file_path} contains the PID of another running process. "
+                    "Configuration issue: edge worker instance must use different PID file paths!"
+                )
+            else:
+                # case 2: previous instance crashed without cleaning up its PID file
+                logger.info("PID file is orphaned. Cleaning up.")
+                pid_file_path.unlink()
+    logger.info("PID file written to %s.", pid_file_path)
```

Comment:
Please log only if this is beneficial for the user
```suggestion
    logger.debug("PID file written to %s.", pid_file_path)
```